src/OpenLoco/Math/Bound.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+#include <algorithm>
+#include <cstdint>
+#include <limits>
+
+namespace OpenLoco::Math::Bound
+{
+    namespace Op
+    {
+        // Bounds an addition to within the capacity of type lhs preventing an overflow
+        // Note: Does not work for 64bit or bigger fields
```

Comment:
I'm not sure if this is possible, but could we static assert on the size of the type?

Reply:
I think we can do `static_assert(sizeof(T) >= 8);`